py/tests/unit/with_runtime_sparkling/test_glrm.py

```diff
@@ -0,0 +1,125 @@
+        bround(dataframe.prediction.getItem(1), precision).alias("prediction1"),
+        bround(dataframe.prediction.getItem(2), precision).alias("prediction2"),
+        bround(dataframe.prediction.getItem(3), precision).alias("prediction3"))
+
+
+def getPreconfiguredAlgorithm():
+    return H2OGLRM(k=4,
+                   transform="Standardize",
+                   loss="Quadratic",
+                   gammaX=0.5,
+                   gammaY=0.3,
+                   seed=42,
+                   recoverSvd=True)
+
+
+def testPipelineSerialization(arrestsDataset):
+    [traningDataset, testingDataset] = arrestsDataset.randomSplit([0.9, 0.1], 42)
+    algo = getPreconfiguredAlgorithm()
+
+    pipeline = Pipeline(stages=[algo])
+    pipeline.write().overwrite().save("file://" + os.path.abspath("build/glrm_pipeline"))
+    loadedPipeline = Pipeline.load("file://" + os.path.abspath("build/glrm_pipeline"))
+    model = loadedPipeline.fit(traningDataset)
+    expected = roundPredictions(model.transform(testingDataset), 7)
+
+    model.write().overwrite().save("file://" + os.path.abspath("build/glrm_pipeline_model"))
+    loadedModel = PipelineModel.load("file://" + os.path.abspath("build/glrm_pipeline_model"))
+    result = roundPredictions(loadedModel.transform(testingDataset), 7)
+
+    unit_test_utils.assert_data_frames_are_identical(expected, result)
+
+
+def testUserXHasEffectOnTrainedModel(spark, arrestsDataset):
+    referenceAlgo = getPreconfiguredAlgorithm()
+    referenceModel = referenceAlgo.fit(arrestsDataset)
+    reference = referenceModel.transform(arrestsDataset)
+
+    column = [5.412, 65.24, -7.54, -0.032, 2.212, 92.24, -17.54, 23.268, 0.312, 123.24, 14.46, 9.768, 1.012, 19.24,
+              -15.54, -1.732, 5.412, 65.24, -7.54, -0.032, 2.212, 92.24, -17.54, 23.268, 0.312, 123.24, 14.46,
+              9.76, 1.012, 19.24, -15.54, -1.732, 5.412, 65.24, -7.54, -0.032, 2.212, 92.24, -17.54, 23.268, 0.312,
+              123.24, 14.46, 9.768, 1.012, 19.24, -15.54, -1.732, 5.412, 65.24]
+    userXData = map(lambda value: (value, value, value, value), column)
+    userXDataFrame = spark.createDataFrame(userXData, ['a', 'b', 'c', 'd'])
+    algo = getPreconfiguredAlgorithm()
+    algo.setUserX(userXDataFrame)
+    algo.setInit("User")
+    model = algo.fit(arrestsDataset)
+    result = model.transform(arrestsDataset)
+
+    unit_test_utils.assert_data_frames_have_different_values(reference, result)
+
+
+def testUserYHasEffectOnTrainedModel(spark, arrestsDataset):
+    referenceAlgo = getPreconfiguredAlgorithm()
+    referenceModel = referenceAlgo.fit(arrestsDataset)
+    reference = referenceModel.transform(arrestsDataset)
+
+    userYData = [(5.412,  65.24,  -7.54, -0.032),
+                 (2.212,  92.24, -17.54, 23.268),
+                 (0.312, 123.24,  14.46,  9.768),
+                 (1.012,  19.24, -15.54, -1.732)]
+    userYDataFrame = spark.createDataFrame(userYData, ['a', 'b', 'c', 'd'])
```

Comment:
This PR proposes to exposes userX and userY as a dataframe. An alternative could be a DenseMatrix from mllib, but I assume that both matrices could be quite large. WDYT?

Reply:
I like the data frame approach!